patches/node/add_should_read_node_options_from_env_option_to_disable_node_options.patch

```diff
@@ -0,0 +1,68 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Samuel Attard <samuel.r.attard@gmail.com>
+Date: Wed, 21 Jul 2021 13:40:59 -0700
+Subject: add should_read_node_options_from_env option to disable NODE_OPTIONS
+ parsing at runtime
+
+We can remove the NODE_OPTIONS environment variable but it in theory could be injected / re-inserted at runtime and be used for workers.  In order to ensure the fuse is respected we need a hard runtime toggle for NODE_OPTION support.
+
+diff --git a/src/env.cc b/src/env.cc
+index 1cc7da1ce15f43905ce607adcc1a23ed9d92948a..613fa9a5e2ea5b6cb3a53110dec437f0aadbdf3f 100644
+--- a/src/env.cc
++++ b/src/env.cc
+@@ -329,6 +329,9 @@ std::string GetExecPath(const std::vector<std::string>& argv) {
+   return exec_path;
+ }
+ 
++/* static */
++bool Environment::should_read_node_options_from_env = true;
++
+ Environment::Environment(IsolateData* isolate_data,
+                          Isolate* isolate,
+                          const std::vector<std::string>& args,
+diff --git a/src/env.h b/src/env.h
+index 45210f074a0ca4d57f9fdc5019e8e82540b28b72..33dad5ec51b7bc8ec21e2fb3243da470ad357641 100644
+--- a/src/env.h
++++ b/src/env.h
+@@ -1143,6 +1143,8 @@ class Environment : public MemoryRetainer {
+   inline double trigger_async_id();
+   inline double get_default_trigger_async_id();
+ 
++  static bool should_read_node_options_from_env;
```

Comment:
_nits_

- This should have a trailing underscore to match existing practice in this class

- This should be declared in Environment's field section along with its other fields

- `_from_env` seems redundant for an Environment variable

Reply:
> This should be declared in Environment's field section along with its other fields

I put it in a spot that had the least frequent changes 😆  Gaming the patch conflict system here

> _from_env seems redundant for an Environment variable

Given node reads NODE_OPTIONS from multiple places and then consolidates into a single set of options I think specifying this flag only disables reading NODE_OPTIONS from env but doesn't disable the CLI parsing is good.